R/assertions.R

```diff
@@ -1698,27 +1700,29 @@ assert_one_to_one <- function(dataset, vars1, vars2) {
 #'   dataset = my_data,
 #'   var = USUBJID
 #' ))
-assert_date_var <- function(dataset, var, dataset_name = NULL, var_name = NULL) {
+assert_date_var <- function(dataset,
+                            var,
+                            dataset_name = rlang::caller_arg(dataset),
+                            var_name = rlang::caller_arg(var),
+                            message = NULL,
+                            class = "assert_date_var",
+                            call = parent.frame()) {
   var <- assert_symbol(enexpr(var))
   assert_data_frame(dataset, required_vars = exprs(!!var))
-  assert_character_scalar(dataset_name, optional = TRUE)
-  assert_character_scalar(var_name, optional = TRUE)
+  assert_character_scalar(dataset_name)
```

Comment:
I think removing the optional = TRUE is good, but wondering why we had it that way?  @bundfussr @zdz2101 do you remember?

Reply:
In the previous iteration the default for the `dataset_name` argument was `NULL`, and we checked that is was indeed a string if a value was passed. In this update, we have a default value that should be a string.

Reply:
@bms63 I think your request makes perfect sense in a simple situation like `mtcars |> foo()`: you would like to see `mtcars`.

But we can quickly get into some ambiguous situations:

```r
mtcars %>% foo()                        # here the data frame is named `.`
mtcars %>% filter(cyl == 6) %>% foo()   # the data frame is again named `.`
mtcars |> filter(dyl == 6) |> foo()     # here the data frame is named `filter(mtcars, cyl == 6)`
```

What do you think? Should we just stick with the argument name? Then it's always defined.